### ISSUE TITLE
Cancel Function

### ISSUE THREAD

**Issue by Sidmaz666** [icefee/hls2mp4#5]
Is there any method to cancel the download?

**Comment by icefee** [comment]
Not yet, but the feature will be added in a future release.

**Comment by Sidmaz666** [comment]
That would be great, also features like pausing would be nice. Can we like extend the Hls2Mp4 class and add these features?

**Comment by icefee** [comment]
Yes, You may need to cache a request list and clean them up promptly when needed

**Comment by icefee** [comment]
Yes, You may need to cache a request list and clean them up promptly when needed.

**Comment by Sidmaz666** [comment]
Cache request list? I don't get it like store the set of request lists in memory?

**Comment by Sidmaz666** [comment]
Can't I just export the destroy method and do something like `Hls2Mp4Instance.destroy()`

**Comment by Sidmaz666** [comment]
I ended up using a flag and throwing a error to stop the download [hls2mp4.js](https://github.com/Sidmaz666/aniswim-website/blob/nextjs/src/components/utils/hls2mp4.js)

**Comment by icefee** [comment]
I have been busy with other things during this period of time, the current design may find it difficult to implement certain features, such as canceling and pausing. I will redesign some key logic in the next version to facilitate the addition of these features. sorry for this. 💔

**Comment by icefee** [comment]
Keep the issue open until this feature has be implemented.

**Comment by Sidmaz666** [comment]
No problem take your time.

**Comment by AmanexDev** [comment]
Do you have any suggestion on how to make a temporary cancel function. I hope this feature will still be added
